connectivity/check/metricssource.go

```diff
@@ -0,0 +1,56 @@
+package check
+
+import (
+	"fmt"
+
+	"github.com/cilium/cilium/pkg/components"
+)
+
+const prometheusContainerPortName = "prometheus"
+
+// MetricsSource defines the info for a source to be used in metrics collection.
+type MetricsSource struct {
+	Name string // the name of the source, e.g.: cilium-agent
+	Pods []Pod  // the list of pods for the given source
+	Port string // the container port value for prometheus
+}
+
+// CiliumAgentMetrics returns the MetricsSource for the cilium-agent component.
+func (ct *ConnectivityTest) CiliumAgentMetrics() MetricsSource {
+	ciliumPods := ct.CiliumPods()
+	if len(ciliumPods) == 0 {
+		return MetricsSource{}
+	}
+
+	var pods []Pod
+	for _, p := range ciliumPods {
+		pods = append(pods, p)
+	}
+
+	source := MetricsSource{
+		Name: components.CiliumAgentName,
+		Pods: pods,
+	}
+
+	// Retrieve the container port value for Prometheus.
+	for _, p := range ciliumPods {
+		// parse all the containers
+		for _, c := range p.Pod.Spec.Containers {
+			if c.Name == components.CiliumAgentName {
+				// parse all the container ports
+				for _, port := range c.Ports {
+					if port.Name == prometheusContainerPortName {
+						source.Port = fmt.Sprintf("%d", port.ContainerPort)
+					}
+				}
+			}
+		}
+	}
+
+	// Prometheus port was not find, let's return an empty MetricsSource.
+	if source.Port == "" {
+		return MetricsSource{}
```

Comment:
Consider defining a `EmptyMetricsSource` constant for this, so callers can compare to the constant to determine if the metrics source exists or not.

Reply:
Cannot define a constant on a non-primitive type, should I create a function?

Reply:
Done! Using a function on MetricsSource.